src/particles/pusher/BeamParticleAdvance.cpp

```diff
@@ -186,19 +186,66 @@ AdvanceBeamParticlesSlice (
                     + dt * 0.5_rt * charge_mass_ratio * Ezp;
 
                 const amrex::ParticleReal gamma_intermediate_inv = 1._rt / std::sqrt( 1._rt
-                    + ux_intermediate*ux_intermediate*clightsq
-                    + uy_intermediate*uy_intermediate*clightsq
-                    + uz_intermediate*uz_intermediate*clightsq );
+                    + (ux_intermediate*ux_intermediate
+                    +  uy_intermediate*uy_intermediate
+                    +  uz_intermediate*uz_intermediate)*inv_c2 );
 
-                const amrex::ParticleReal uz_next = uz + dt * charge_mass_ratio
+                amrex::ParticleReal uz_next = uz + dt * charge_mass_ratio
                     * ( Ezp + ( ux_intermediate * Byp - uy_intermediate * Bxp )
                     * gamma_intermediate_inv );
 
+                if (radiation_reaction) {
+
+                    const amrex::ParticleReal Exp = ExmByp + clight*Byp;
+                    const amrex::ParticleReal Eyp = EypBxp - clight*Bxp;
+
+                    const amrex::ParticleReal gamma_intermediate = std::sqrt( 1._rt
+                        + (ux_intermediate*ux_intermediate
+                        +  uy_intermediate*uy_intermediate
+                        +  uz_intermediate*uz_intermediate)*inv_c2 );
+                    // Estimation of the velocity at intermediate time
+                    const amrex::ParticleReal vx_n = ux_intermediate*gamma_intermediate_inv;
+                    const amrex::ParticleReal vy_n = uy_intermediate*gamma_intermediate_inv;
+                    const amrex::ParticleReal vz_n = uz_intermediate*gamma_intermediate_inv;
+                    const amrex::ParticleReal bx_n = vx_n*inv_clight;
+                    const amrex::ParticleReal by_n = vy_n*inv_clight;
+                    const amrex::ParticleReal bz_n = vz_n*inv_clight;
+
+                    // Lorentz force over charge
+                    const amrex::ParticleReal flx_q = (Exp + vy_n*Bzp - vz_n*Byp);
+                    const amrex::ParticleReal fly_q = (Eyp + vz_n*Bxp - vx_n*Bzp);
+                    const amrex::ParticleReal flz_q = (Ezp + vx_n*Byp - vy_n*Bxp);
+                    const amrex::ParticleReal fl_q2 = flx_q*flx_q + fly_q*fly_q + flz_q*flz_q;
+
+                    // Calculation of auxiliary quantities
+                    const amrex::ParticleReal bdotE = (bx_n*Exp + by_n*Eyp + bz_n*Ezp);
+                    const amrex::ParticleReal bdotE2 = bdotE*bdotE;
+                    const amrex::ParticleReal coeff = gamma_intermediate*gamma_intermediate*(fl_q2-bdotE2);
+
+                    // Radiation reaction constant
+                    const amrex::ParticleReal q_over_mc = charge_mass_ratio*inv_clight;
+                    const amrex::ParticleReal RRcoeff = (2.0_rt/3.0_rt)*PhysConstSI::r_e*q_over_mc*q_over_mc;
```

Comment:
Can you calculate RRcoeff outside of the kernel and also make it work in normalized units using `plasmas.background_density_SI` ?

Reply:
I am going to add normalized units in a separate PR, since this will require shifting the `plasmas.background_density_SI` to `hipace.background_density_SI` and I found this to be cleaner in a separate PR.